app/com/linkedin/drelephant/tuning/hbt/MRApplicationData.java

```diff
@@ -0,0 +1,742 @@
+  }
+
+  /**
+   * Add reducer suggested parameter to Map
+   * @param heapSizeMax
+   * @param containerSize
+   * @param heuristicsResultID
+   */
+  private void addReducerMemoryAndHeapToSuggestedParameter(Double heapSizeMax, Double containerSize,
+      String heuristicsResultID) {
+    suggestedParameter.put(REDUCER_MEMORY_HADOOP_CONF.getValue(), containerSize);
+    suggestedParameter.put(REDUCER_HEAP_HADOOP_CONF.getValue(), heapSizeMax);
+    logDebuggingStatement(" Memory Assigned " + heuristicsResultID + "_Reducer " + suggestedParameter.get(
+        REDUCER_MEMORY_HADOOP_CONF.getValue()),
+        " Heap Assigned " + heuristicsResultID + "_Reducer " + suggestedParameter.get(
+            REDUCER_HEAP_HADOOP_CONF.getValue()));
+  }
+
+  /**
+   * todo : currently task for reducers are not processed  , since we have to find
+   * out the way to pass number of reducer tasks from azkaban
+   * @param yarnAppHeuristicResult
+   * @param functionType
+   */
+  private void processForNumberOfTask(AppHeuristicResult yarnAppHeuristicResult, String functionType) {
+    if (functionType.equals(Mapper.name())) {
+      processForNumberOfTaskMapper(yarnAppHeuristicResult);
+    } /*else if (functionType.equals(Reducer.name())) {
+      processForNumberOfTaskReducer(yarnAppHeuristicResult);
+    }*/
+  }
+
+  /**
+   * IF the mapper time heuristics is failed , its failed because either the
+   * average input going to one task is too small or too large .
+   * If its too small , increase the average task input size and that would be the
+   * split size . If time is too big , decrease the average task input size and that would
+   * be split size.
+   * @param yarnAppHeuristicResult : Heuristic details about the application
+   * @return : New split size
+   */
+  private void processForNumberOfTaskMapper(AppHeuristicResult yarnAppHeuristicResult) {
+    logger.info("Calculating Split Size ");
+    Map<String, String> heuristicsResults = yarnAppHeuristicResult.yarnAppResult.getHeuristicsResultDetailsMap();
+    double averageTaskInputSize = getPreviousAverageTaskInputSize(heuristicsResults);
+    double averageTaskTimeInMinute = getPreviousAverageTaskRunTime(heuristicsResults);
+    addCounterData(
+        new String[]{Mapper + " " + AVERAGE_TASK_INPUT_SIZE.getValue(), Mapper + " " + AVERAGE_TASK_RUNTIME.getValue()},
+        averageTaskInputSize, averageTaskTimeInMinute);
+    if (isSplitSizePrerequisiteQualify(averageTaskInputSize, averageTaskTimeInMinute)) {
+      calculateAndUpdateNewValuesForSplitSize(averageTaskInputSize, averageTaskTimeInMinute);
+    }
+  }
+
+  /**
+   * Calcualte and update new split size
+   * @param averageTaskInputSize
+   * @param averageTaskTimeInMinute
+   */
+  private void calculateAndUpdateNewValuesForSplitSize(double averageTaskInputSize, double averageTaskTimeInMinute) {
+    long newSplitSize = 0L;
+    if (averageTaskTimeInMinute <= AVG_TASK_TIME_LOW_THRESHOLDS_FIRST.getValue()) {
```

Comment:
actually its used twice ,since it have value 2. But one time it will be used to increase split size and other time it will be used to decrease split size.